crates/ruff/src/rules/pylint/helpers.rs

```diff
@@ -44,3 +45,52 @@ pub(crate) fn in_dunder_init(model: &SemanticModel, settings: &Settings) -> bool
     }
     true
 }
+
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+pub(crate) enum ViolationsCmpop {
+    Eq,
+    NotEq,
+    Lt,
+    LtE,
+    Gt,
+    GtE,
+    Is,
+    IsNot,
+    In,
+    NotIn,
+}
+
+impl From<&ast::Cmpop> for ViolationsCmpop {
+    fn from(cmpop: &ast::Cmpop) -> Self {
+        match cmpop {
+            ast::Cmpop::Eq => Self::Eq,
+            ast::Cmpop::NotEq => Self::NotEq,
+            ast::Cmpop::Lt => Self::Lt,
+            ast::Cmpop::LtE => Self::LtE,
+            ast::Cmpop::Gt => Self::Gt,
+            ast::Cmpop::GtE => Self::GtE,
+            ast::Cmpop::Is => Self::Is,
+            ast::Cmpop::IsNot => Self::IsNot,
+            ast::Cmpop::In => Self::In,
+            ast::Cmpop::NotIn => Self::NotIn,
+        }
+    }
+}
+
+impl fmt::Display for ViolationsCmpop {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let representation = match self {
+            Self::Eq => "==",
+            Self::NotEq => "!=",
+            Self::Lt => "<",
+            Self::LtE => "<=",
+            Self::Gt => ">",
+            Self::GtE => ">=",
+            Self::Is => "is",
+            Self::IsNot => "is not",
+            Self::In => "in",
+            Self::NotIn => "not in",
+        };
+        write!(f, "{representation}")
+    }
+}
```

Comment:
Do you need the `ViolationsCmpop` only to implement `Display`? An alternative design that uses fewer code is to wrap the `Cmpop` instead
```suggestion
pub(crate) struct ViolationsCmpop(Cmpop);

impl fmt::Display for ViolationsCmpop {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let representation = match self.0 {
            Self::Eq => "==",
            Self::NotEq => "!=",
            Self::Lt => "<",
            Self::LtE => "<=",
            Self::Gt => ">",
            Self::GtE => ">=",
            Self::Is => "is",
            Self::IsNot => "is not",
            Self::In => "in",
            Self::NotIn => "not in",
        };
        write!(f, "{representation}")
    }
}
```

Reply:
Yes, this looks better.

Reply:
Wouldn't we still need to implement `From<&Cmpop>` etc. for the `comparison-of-constants` and `comparison-with-itself` rules which are passed `&[Cmpop]`? The CI complained about conversions not being implemented.